### PR TITLE
AP-59-Calendar-Validation

### PR DESCRIPTION
**Pull request by FreshClouds**
Created calendar validation

Checks only for current and future events will allow saving of historic event despite that not sounding like a best practice. 

Different commits contain comments on what I was trying and what happened. Including links.

### CODE REVIEW

**Review by FreshClouds — COMMENTED** [review]
Ready for review, weekend check is also working and included in its own test method.

**Review by off-nadir-insight — APPROVED** [review]
code looks good with updates discussed around the `isWeekend` check. Clear to push. Only follow up would be to explore the inner-class constructors. Good stretch goal!

File: force-app/main/default/classes/EventTriggerHandler.cls [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        }
+        return assignedIdToEvents;
+    }
+    private class DatesToCompare{
+        private Datetime startDateTimeOne;
+        private Datetime endDateTimeOne;
+        private Datetime startDateTimeTwo;
+        private DateTime endDateTimeTwo;
+
+        // private DatesToCompare(Datetime startDateOne, Datetime endDateOne, Datetime startDateTwo, Datetime endDateTwo){
```
Comment (fritzb100) [inline force-app/main/default/classes/EventTriggerHandler.cls:55]: looks like this is commented out. Are you still working on this?

File: force-app/main/default/classes/EventTriggerHandler.cls [hunk 2]
```diff
@@ -0,0 +1,62 @@
+public with sharing class EventTriggerHandler {
+    public static void preventOverlappingFutureEventsForAssignedUser(List<Event> triggerDotNew) {
+        Map<Id, List<Event>> ownerAssignedIdToEvents = createAssignedIdToEventListMap(triggerDotNew);
+        for(Event ev: triggerDotNew){
```
Comment (fritzb100) [inline force-app/main/default/classes/EventTriggerHandler.cls]: Would it be better for the variable to be a little longer than one or two letters, I know the instructors weren't too keen on that one.

**Review by fritzb100 — APPROVED** [review]

**Pull request merged**